jgrapht-core/src/main/java/org/jgrapht/alg/shortestpath/CHManyToManyShortestPaths.java

```diff
@@ -86,12 +87,25 @@
      * Constructs an instance of the algorithm for a given {@code graph}.
      *
      * @param graph a graph
+     * @deprecated replaced with {@link #CHManyToManyShortestPaths(Graph, ThreadPoolExecutor)}
      */
+    @Deprecated
     public CHManyToManyShortestPaths(Graph<V, E> graph)
```

Comment:
Instead of deprecating, I would provide a default using `newFixedThreadPool(Runtime.getRuntime().availableProcessors())` but of course without explicitly closing anything at the end of the algorithm.  Same comment in the other places as well.

Reply:
Sorry for the late response. I agree that the deprecation is not necessary here. However, I\`ve explicitly specified in the documentation that some of the constructors do not provide a possibility to shut down the executors at the end of the algorithm.